### PR TITLE
Componentize markup and stop partials from piping into `dist`

### PR DESCRIPTION
**Pull request by CodeTheory**
@matheuss @sindresorhus @doot0 Please review if anyone gets a chance 🙌 

This PR works to shrink the amount of repeated pug code in Kap by introducing mixins. It also stops any `.pug` or `.css` file starting with `_` from piping into the `dist` directory. Handy for partials! 🎉

### CODE REVIEW

File: app/src/renderer/views/_mixins.pug [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  - Re-usable components in the kap markup
+
+
+//- **
+    name: icon
+    description: An easier method for fetching SVGs in markup.
+    @param {string} name - Name of the svg file you're calling
+    example: +icon('crop')
+mixin icon(name)
+  img(src='/app/static/'+name+'.svg')&attributes(attributes)
```
Comment (sindresorhus) [inline app/src/renderer/views/_mixins.pug]: Not sure whether you can use template literals in Pug templates, but would look nicer if possible:

```js
img(src=`/app/static/${name}.svg`)&attributes(attributes)
```

File: app/src/renderer/views/_extend.pug [hunk 2]
```diff
@@ -0,0 +1,15 @@
+include _mixins
+
+doctype html
+html(lang='en')
```
Comment (sindresorhus) [inline app/src/renderer/views/_extend.pug]: Consistent quoting. Pick either single or double quotes.

File: app/src/renderer/views/_extend.pug [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  head
+    meta(charset="UTF-8")
```
Comment (sindresorhus) [inline app/src/renderer/views/_extend.pug]: UTF-8 => utf-8

File: app/src/renderer/views/_mixins.pug [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+
+//- **
+    name: window-header
+    description: Re-usable window header for each window.
+    attributes:
+      - {string}  title               - The name of the window
+      - {boolean} [invisibleControls] - adds a class to the window controls that gives them full transparency
```
Comment (sindresorhus) [inline app/src/renderer/views/_mixins.pug]: adds => Adds

File: app/src/renderer/views/_mixins.pug [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        .c-range-label.is-middle.js-middle-label=a.middleLabel
+        .c-range-label=a.highLabel
+
+//- **
+    name: stylesheet
+    description: Easier referencing of stylesheets
+    @param {string} name - name of the stylesheet to fetch
+    example: +stylesheet('cropper')
+mixin stylesheet(name)
+  link(href='../css/' + name + '.css' rel='stylesheet' type='text/css')
```
Comment (sindresorhus) [inline app/src/renderer/views/_mixins.pug]: `type='text/css'` is moot as it's the default, so can be dropped.

**Comment by sindresorhus** [comment]
I'm getting:

> utils.js:22 Uncaught TypeError: Cannot read property 'addEventListener' of null

In the GIF editor.

**Comment by sindresorhus** [comment]
The cropper window is broken. It's not possible to change its height.

**Comment by matheuss** [comment]
@CodeTheory the marquee is smaller than the window 😰 
screen shot 2016-12-09 at 2 58 52 pm

**Comment by CodeTheory** [comment]
@matheuss @sindresorhus please review again ❤️

**Pull request merged**